src/App.tsx

```diff
@@ -15,48 +16,119 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  SORT_BY_LENGTH = 'length',
+  SORT_BY_ALPHABET = 'alpha',
+  initialSortField = '',
+}
+
+function sortBy(fieldForSort: SortType, isReversed: boolean): string[] {
+  const goods = [...goodsFromServer];
+
+  if (fieldForSort) {
+    switch (fieldForSort) {
+      case SortType.SORT_BY_ALPHABET:
+        goods.sort((a, b) => a.localeCompare(b));
+        break;
+
+      case SortType.SORT_BY_LENGTH:
+        goods.sort((a, b) => a.length - b.length);
+        break;
+
+      default:
+        break;
+    }
+  }
+
+  return isReversed ? goods.reverse() : goods;
+}
+
 export const App: React.FC = () => {
+  const [isReverseActive, setIsReverseActive] = useState(false);
+  const [sortField, setSortField] = useState(SortType.initialSortField);
+
+  const preparedGoods: string[] = sortBy(sortField, isReverseActive);
+
+  const resetAllSort = (): void => {
```

Comment:
it's a good idea to put this in a separate function instead of doing it on a button.